lib/spack/spack/mirror.py

```diff
@@ -98,6 +98,63 @@ def from_dict(d, name=None):
         else:
             return Mirror(d['fetch'], d['push'], name)
 
+    @staticmethod
+    def from_args(args):
```

Comment:
The purpose here is to construct a valid mirror given a set of user-supplied options, which is something different from what `from_dict` is meant to do.  At the moment, there is only one call site in `cmd/mirror.py` but that will soon change to include other commands that need to accept mirror arguments. (e.g.: #25765)

The main value of the method is the validation and lookup logic, and the intent of factoring it out is to avoid the need for multiple commands to duplicate this logic.  The coupling with spack cmd code is intentional, since that is where the intended call sites would be.

I think the right call is to have `from_args` take keyword arguments instead of an object with attributes, which should simplify unit testing.